x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plugin/SqlMediaTypeParserTests.java

```diff
@@ -0,0 +1,110 @@
+    }
+
+    public void testMediaTypeDetectionWithParameters() {
+        assertThat(getResponseMediaType(reqWithAccept("text/plain; charset=utf-8"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(PLAIN_TEXT));
+        assertThat(getResponseMediaType(reqWithAccept("text/plain; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(PLAIN_TEXT));
+        assertThat(getResponseMediaType(reqWithAccept("text/plain; charset=utf-8; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(PLAIN_TEXT));
+
+        assertThat(getResponseMediaType(reqWithAccept("text/csv; charset=utf-8"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(CSV));
+        assertThat(getResponseMediaType(reqWithAccept("text/csv; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(CSV));
+        assertThat(getResponseMediaType(reqWithAccept("text/csv; charset=utf-8; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(CSV));
+
+        assertThat(getResponseMediaType(reqWithAccept("text/tab-separated-values; charset=utf-8"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(TSV));
+        assertThat(getResponseMediaType(reqWithAccept("text/tab-separated-values; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(TSV));
+        assertThat(getResponseMediaType(reqWithAccept("text/tab-separated-values; charset=utf-8; header=present"),
+            createTestInstance(false, Mode.PLAIN, false)).textFormat(), is(TSV));
+    }
+
+    public void testInvalidFormat() {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class,
+            () -> getResponseMediaType(reqWithAccept("text/garbage"), createTestInstance(false, Mode.PLAIN, false)));
+        assertEquals(e.getMessage(), "invalid format [text/garbage]");
+    }
+
+    public void testNoFormat() {
+        IllegalArgumentException e = expectThrows(IllegalArgumentException.class,
+            () ->  getResponseMediaType(new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY).build(),
+                createTestInstance(false, Mode.PLAIN, false)));
+        assertEquals(e.getMessage(), "Invalid request content type: Accept=[null], Content-Type=[null], format=[null]");
+    }
+
+    private static RestRequest reqWithAccept(String acceptHeader) {
+
+        return new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+            .withHeaders(new HashMap<String, List<String>>() {{
+                put("Content-Type", Collections.singletonList("application/json"));
+                put("Accept", Collections.singletonList(acceptHeader));
+            }}).build();
+    }
+
+    protected SqlQueryRequest createTestInstance(boolean binaryCommunication, Mode mode, boolean columnar) {
```

Comment:
binaryCommunications seems to be always `false`?

Reply:
And so the rest of the parameters were always having the same value. I've removed them.